### PR TITLE
Additional improvements to the monitoring library

### PR DESCRIPTION
**Pull request by andrewrlee**
* Type checking tests - adding a step to build to ensure types are consistent across tests and production code
 * Adding initial draft of a README
 * Rename components arg -> healthComponents

### CODE REVIEW

File: packages/monitoring/package.json [hunk 1]
```diff
@@ -20,7 +20,7 @@
     "node": "20 || 22"
   },
   "scripts": {
-    "build": "rollup -c rollup.config.ts --bundleConfigAsCjs",
+    "build": "tsc && rollup -c rollup.config.ts --bundleConfigAsCjs",
```
Comment (ushkarev) [inline packages/monitoring/package.json:23]: maybe type checking is more of a _test_ action? (which is a prerequisite of publishing a release so would be certain to happen in CI)

Comment (andrewrlee) [reply packages/monitoring/package.json:23]: Can do, it's weird as `build` is more of a "produce artifact" task rather than "build the product"

Comment (andrewrlee) [reply packages/monitoring/package.json]: that's done now

Comment (andrewrlee) [reply packages/monitoring/package.json]: would linting also fit in to that too?

Comment (ushkarev) [reply packages/monitoring/package.json]: CI does `npm test` and `npm run lint` in parallel for all runs. but not `npm run build`; that only happens on release

Comment (andrewrlee) [reply packages/monitoring/package.json]: I can fix that after a release

Comment (ushkarev) [reply packages/monitoring/package.json]: how about this? https://github.com/ministryofjustice/hmpps-typescript-lib/pull/25

**Pull request merged**